mongoengine/base/datastructures.py

```diff
@@ -188,7 +191,10 @@ def _mark_as_changed(self, key=None):
 class EmbeddedDocumentList(BaseList):
     def __init__(self, list_items, instance, name):
         super().__init__(list_items, instance, name)
-        self._instance = instance
```

Comment:
This is where the problem was, although self._instance was correctly converted to a weakref.proxy in `super().__init__`, it was being overwritten here.

This code is actually unnecessary since it's already handled in the parent class `__init__`. I'll push an updated version of this code (getting rid of EmbeddedDocumentList.__init__ as it's not needed)